Ng Wei Hong iP

justweihong opened the pull request.

Commit:
Add Gradle support

justweihong committed:
finished level 1

justweihong committed:
finished level-2

justweihong committed:
done with level-3

justweihong committed:
finished level-4, but did not use inheritance and polymorphism

justweihong committed:
finished level-5, oh but i didnt see the automated Text UI Testing

justweihong committed:
done automated testing

justweihong committed:
finish level-6

justweihong committed:
incorporated inheritance of Tasks: EventTask, DeadlineTask and TodoTask classes

justweihong committed:
configured filewriter to change file given new taskList

justweihong committed:
added updateFiles for other taskList changes methods, pending more OOP changes for main method of Duke

justweihong committed:
Merge pull request #1 from justweihong/branch-Level-7

Branch level 7 (pending oop enhancement before finalising level 7)

justweihong committed:
Configure Storage class and refactor Duke main method

Storage class to handle hard disk storage, including existing and
non-existing text file.

File storage method not working because of static reference in main
method. Replace code with individual run() method for non-static
context.

justweihong committed:
Configure DukeDate class

Previously stored due dates for Events and Deadlines in strings. This
class creates a custom template to display these due dates.

justweihong committed:
Merge branch 'branch-Level-7'

justweihong committed:
merged 'master' with branch-Level-8

justweihong committed:
Separate main method to Command, Ui, Storage, Parser and TaskList
classes.

Command classes include inherited commands (add, delete, done, exit and
list) so it is more open to extension.

Parser handles all user input while Ui handles all output that prints.
TaskList handles additional methods with print and set task as done.
Storage continues to handle hard drive file duke.txt

aizatazhar commented on src/main/java/AddCommand.java:
```diff
@@ -0,0 +1,37 @@
+        String due;
+        Task newTask;
+        switch (taskType) {
+            case "todo":
+                newTask = new TodoTask(description);
+                break;
+            case "deadline":
+                due = parsedStrings[2];
+//                System.out.println(description);
+//                System.out.println(due);
+                newTask = new DeadlineTask(description, due);
+                break;
+            case "event":
+                due = parsedStrings[2];
+                newTask = new EventTask(description, due);
+                break;
+            default:
+                throw new DukeException("I don't know what type of task this is :(");
```
I believe case clauses should not be indented

aizatazhar commented on src/main/java/DeadlineTask.java:
```diff
@@ -0,0 +1,21 @@
+    }
+
+    @Override
+    public String fileString() {return "D | " + super.fileString() + " | " + due; }
```
I think egyptian style braces should be used here

aizatazhar commented on src/main/java/DukeDate.java:
```diff
@@ -0,0 +1,23 @@
+    private LocalDate localDate;
+
+    public DukeDate(String datetimeString) throws DukeException {
+        String[] datetimeArray = datetimeString.split(" ");
```
I'm not sure about this but should the variable names be dateTimeArray and dateTimeString instead?

aizatazhar commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,69 @@
+                return new ListCommand();
+            case "bye":
+                return new ExitCommand();
+            case "todo":
+            case "deadline":
```
Perhaps you can consider placing 2 comments here to be clear that fallthrough is intentional here?

aizatazhar commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,94 @@
+        this.file = new File(filePath);
+    }
+
+    /** Helper: creates file if it is not present in filePath.
+     * @throws IOException
+     */
```
Here the Javadocs opening /** should be on a separate line and there should be a blank line between the description and parameter section

aizatazhar commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,94 @@
+    }
+
+    /** Helper: Converts text line to a Task object.
+     * @oaram fileString A single line string from file text
```
small typo (should be @param instead), and every parameter description should end with a period

aizatazhar commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,94 @@
+    /** Helper: generate the Tasks based on the text file.
+     * @return TaskList that contains the collection of Tasks in the storage.
+     */
+    public TaskList generateTaskList () throws DukeException {
```
Theres an extra whitespace at the function name

aizatazhar commented on src/main/java/TodoTask.java:
```diff
@@ -0,0 +1,18 @@
+    }
+
+    @Override
+    public String fileString() {return "T | " + super.fileString(); }
```
I think egyptian brackets should be used here

aizatazhar reviewed (commented):
Great job! There's a few more switch statements I noticed where the case clauses should be indented as well as for Javadocs

justweihong committed:
configure JUnit test for Parser and EventTask classes

This is to reduce the need for running whole Duke for easier testing purposes.

justweihong committed:
include javadoc comments for classes

justweihong committed:
Merge branch 'master' of https://github.com/justweihong/ip

justweihong committed:
package files into subfolders so it is more organised

justweihong committed:
remove unnecessary comments and wildcard imports, slight mofification of
code structure for coding standard

justweihong committed:
configure find command to search for tasks that matches description

justweihong committed:
Merge pull request #3 from justweihong/branch-Level-9

configure find command to search for tasks that matches description

justweihong committed:
Merge branch 'master' into add-gradle-support

justweihong committed:
Merge pull request #4 from justweihong/add-gradle-support

Add gradle support

justweihong committed:
add GUI for duke project

justweihong committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	build.gradle
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

justweihong committed:
add assertions

justweihong committed:
Merge pull request #6 from justweihong/branch-A-Assertions

add assertions

justweihong committed:
refactor abstraction layers for code quality

This is in consideration of SLAP principle, avoiding arrowhead codes,
and having guard clauses for file storage.

justweihong committed:
Merge pull request #7 from justweihong/branch-A-CodeQuality

refactor abstraction layers for code quality

justweihong committed:
add priority feature to mark Tasks' priority

justweihong committed:
Merge pull request #8 from justweihong/branch-C-Priority

add priority feature to mark Tasks' priority

justweihong committed:
include JavaDoc branch

justweihong committed:
Merge pull request #9 from justweihong/branch-A-JavaDoc

include JavaDoc branch

justweihong committed:
include CodingStandard branch

justweihong committed:
Merge pull request #10 from justweihong/branch-CodingStandard

include CodingStandard branch

justweihong committed:
style Gui to look better

justweihong committed:
Merge pull request #11 from justweihong/branch-A-BetterGui

style Gui to look better

justweihong committed:
fix indexing bug for done and priority commands

justweihong committed:
Merge branch 'master' into branch-Level-8

# Conflicts:
#	src/main/java - Copy/DeadlineTask.java
#	src/main/java - Copy/EventTask.java
#	src/main/java/DeadlineTask.class
#	src/main/java/data/duke.txt

justweihong committed:
perpare to merge to master

justweihong committed:
Merge pull request #12 from justweihong/branch-Level-8

Branch level 8 (to allow ip progress to track this branch)

justweihong committed:
add Ui picture

Ui picture represents tis product in full glory.

justweihong committed:
fix input bugs for done and delete commands.

Exceptions are thrown when done and delete commands do not include
integers. Fix this bug to respond to user to include an integer.

justweihong committed:
configure platform exit upon exit command

previously window does not close and exit when "bye" is typed as an
input. Added platform exit method to close window when "bye" is typed as
an input.

justweihong committed:
add help command, include user guide assets.

include help commands that list out all commands available for DukeBB.
Include screenshots of commands output as part of user guide.

justweihong committed:
Update README.md

include an introduction, a feature list, features description, format and examples of usage.

justweihong committed:
refactor test classes for gradle support, modify build gradle to build
jar file

test classes have the wrong imports and cause gradle to produce
exceptions during build

justweihong committed:
configure file storage for priority command, configure non-hard-coded
file path for storage

priority marking is not saved when dukeBB closes and will reset to
'nil'. File path shows error when jar file is used on other location.
This commit resolves these two issues.

justweihong committed:
Set theme jekyll-theme-cayman
